test/integration/hotrestart_test.sh

```diff
@@ -45,20 +46,29 @@ if [[ -z "${ENVOY_IP_TEST_VERSIONS}" ]] || [[ "${ENVOY_IP_TEST_VERSIONS}" == "al
   JSON_TEST_ARRAY+=("${HOT_RESTART_JSON_V6}")
 fi
 
+# Enable this test to work with --runs_per_test
+if [[ -z "${TEST_RANDOM_SEED}" ]]; then
+  BASE_ID=1
+else
+  BASE_ID="${TEST_RANDOM_SEED}"
```

Comment:
Arguably this still has the issue that if you run two distinct `bazel test` on the same machine, with different run IDs, you will have potential conflict. I think it's fine for now, this is a useful improvement over status quo.

Reply:
Doesn't Bazel serialize separate build or test invocations made on the same machine in the same code base? When I try it in a separate window it tells me it's waiting for the first command to complete.

Reply:
I think it's per-bazel-daemon.  On an interactive system, that's generally 1 per user (I think).  On a build machine, it's probably 1 per container.

Reply:
It's per source directory actually (I know because I run multiple concurrently in different Envoy trees :).